samples/peptide_mixture_grxmc_ideal.py

```diff
@@ -60,13 +61,17 @@
 parser.add_argument('--no_verbose', action='store_false', help="Switch to deactivate verbose",default=True)
 args = parser.parse_args()
 
-if args.mode not in valid_modes:
-    raise ValueError(f"Mode {args.mode} is not currently supported, valid modes are {valid_modes}")
-
 # The trajectories of the simulations will be stored using espresso built-up functions in separed files in the folder 'frames'
 Path("./frames").mkdir(parents=True, 
                        exist_ok=True)
 
+steps_name = f"{espressomd.version.friendly() == '4.2' and 'reaction_' or ''}steps"
+
+def get_number_of_particles(espresso_system, ptype):
+    args = (ptype,) if espressomd.version.friendly() == "4.2" else ()
+    kwargs = {} if espressomd.version.friendly() == "4.2" else {"type": ptype}
+    return espresso_system.number_of_particles(*args, **kwargs)
```

Comment:
In the same spirit as before maybe expanding the code a bit for better readability?
```python
def get_number_of_particles(espresso_system, ptype):
    """
    Gets the number of particles of a given type `ptype` from espresso, handling changes recent changes in the API between the version 4.2 and 4.3 of espresso.
    """
    if espressomd.version.friendly() == "4.2":
                 args = (ptype,)
                 kwargs = {}
    else:
                 args = ()
                 kwargs = {"type": ptype}
    return espresso_system.number_of_particles(*args, **kwargs)
```
If you agree then the function should also be changed in the other sample scripts for consistency.